examples/graphql-react-apollo/src/setupTests.js

```diff
@@ -3,13 +3,20 @@
 // expect(element).toHaveTextContent(/react/i)
 // learn more: https://github.com/testing-library/jest-dom
 import '@testing-library/jest-dom/extend-expect'
+import { client } from './ApolloClient'
 import { server } from './mocks/server'
 
 beforeAll(() => {
   // Enable the mocking in tests.
   server.listen()
 })
 
+beforeEach(() => {
+  // Ensure Apollo cache is cleared between tests.
+  // https://www.apollographql.com/docs/react/api/core/ApolloClient/#ApolloClient.clearStore
+  return client.clearStore()
```

Comment:
What is the difference between `client.clearStore()` and `client.cache.reset()`?

I've been always using the latter one to ensure there's no cache between tests in Apollo. It also looks like a more semantic thing to do based on the methods called.

Reply:
`clearStore()` resets the QueryManager and cancels any pending queries before clearing the cache (it still calls `cache.reset()` under the hood) https://github.com/apollographql/apollo-client/blob/main/src/core/QueryManager.ts#L698-L724

It seems to be the blessed path for resetting the cache, as `InMemoryCache.reset()` isn't mentioned in the class docs https://www.apollographql.com/docs/react/api/cache/InMemoryCache/

In the case of testing, either is likely to work, so if you'd rather I switch it to `client.cache.reset()` I can do that.

Reply:
Request cancellation sounds great in the `beforeEach` hook! Thanks for explaining this.